lib/rucio/common/utils.py

```diff
@@ -897,39 +897,39 @@ class Color:
 
 def detect_client_location():
     """
-    Open a UDP socket to a machine on the internet, to get the local IPv4 and IPv6
-    addresses of the requesting client.
+    Client IP will be set on the server side (request.remote_addr)
 
     Try to determine the sitename automatically from common environment variables,
     in this order: SITE_NAME, ATLAS_SITE_NAME, OSG_SITE_NAME. If none of these exist
     use the fixed string 'ROAMING'.
+
+    If environment variables sets location, it uses it.
     """
 
     ip = '0.0.0.0'
-    try:
-        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-        s.connect(("8.8.8.8", 80))
-        ip = s.getsockname()[0]
-    except Exception:
-        pass
-
-    ip6 = '::'
-    try:
-        s = socket.socket(socket.AF_INET6, socket.SOCK_DGRAM)
-        s.connect(("2001:4860:4860:0:0:0:0:8888", 80))
-        ip6 = s.getsockname()[0]
-    except Exception:
-        pass
 
     site = os.environ.get('SITE_NAME',
                           os.environ.get('ATLAS_SITE_NAME',
                                          os.environ.get('OSG_SITE_NAME',
                                                         'ROAMING')))
 
+    latitude = os.environ.get('RUCIO_LATITUDE')
+    longitude = os.environ.get('RUCIO_LONGITUDE')
+    if latitude and longitude:
+        try:
+            latitude = float(latitude)
+            longitude = float(longitude)
+        except ValueError:
+            latitude = longitude = 0
```

Comment:
I would not set it to 0,0 - although unlikely, it's still a valid input. Just put `None` in the error/default case.

Reply:
that's what regular GeoIP would give. It is at least defined. But I changed it to None.